llvm/include/llvm/CodeGen/SDPatternMatch.h

```diff
@@ -358,6 +358,23 @@ struct Or<Pred, Preds...> : Or<Preds...> {
   }
 };
 
+template <typename Pred> struct Not {
+  Pred P;
+
+  explicit Not(const Pred &P) : P(P) {}
+
+  template <typename MatchContext>
+  bool match(const MatchContext &Ctx, SDValue N) {
+    return !P.match(Ctx, N);
+  }
+};
+// Explicit deduction guide.
+template <typename Pred> Not(const Pred &P) -> Not<Pred>;
+
+template <typename Pred> inline Not<Pred> m_IsNot(const Pred &P) {
```

Comment:
Is that would PatternMatch.h calls m_Unless?

Reply:
That would indeed be more consistent. It's updated now.